test/src/test/java/org/corfudb/benchmark/SMRMapBenchmark.java

```diff
@@ -0,0 +1,207 @@
+     * Keeps a single map, and a {@link AtomicInteger} which is used as a counter
+     * to issue new keys.
+     */
+    public static class SMRBenchmarkState extends CorfuSingleNodeBenchmarkState {
+
+        /**
+         * A runtime to use during the iteration.
+         */
+        CorfuRuntime rt;
+
+        /**
+         * A map to use during each iteration.
+         */
+        SMRMap<Integer, Integer> map;
+
+        /**
+         * A counter to be used during each iteration.
+         */
+        AtomicInteger counter;
+
+        /**
+         * Initializes a new runtime, map and counter.
+         */
+        @Override
+        public void initializeIteration() {
+            super.initializeIteration();
+
+            rt = getNewRuntime();
+            map = rt.getObjectsView().build()
+                    .setStreamName("test")
+                    .setTypeToken(new TypeToken<SMRMap<Integer, Integer>>() {
+                    })
+                    .open();
+            counter = new AtomicInteger();
+        }
+
+        /**
+         * Get the next key.
+         */
+        public int getNextKey() {
+            return counter.getAndIncrement();
+        }
+    }
+
+
+    /**
+     * Measure the performance of pure insert operations. A single operation is performed per trial.
+     */
+    @Benchmark
+    @Warmup(iterations = 5)
+    @Measurement(iterations = 10)
+    public void pureInsert(SMRBenchmarkState state) {
+        state.map.blindPut(state.getNextKey(), 0);
+    }
+
+    /**
+     * Measure the performance of pure insert operations multithreaded. A single operation is performed
+     * per trial.
+     */
+    @Benchmark
+    @Warmup(iterations = 5)
+    @Measurement(iterations = 10)
+    @Threads(5)
+    public void pureInsertMultiThreaded(SMRBenchmarkState state) {
+        pureInsert(state);
+    }
+
+    /**
+     * Measure the performance of insert operations. A single operation is performed per trial.
+     */
+    @Benchmark
+    @Warmup(iterations = 5)
+    @Measurement(iterations = 10)
+    public void insert(SMRBenchmarkState state) {
+        state.map.put(state.getNextKey(), 0);
+    }
+
+    /**
+     * Measure the performance of insert operations multithreaded. A single operation is performed
+     * per trial.
+     */
+    @Benchmark
+    @Warmup(iterations = 5)
+    @Measurement(iterations = 10)
+    @Threads(5)
+    public void insertMultiThreaded(SMRBenchmarkState state) {
+        insert(state);
+    }
+
+    /**
+     * A more complex benchmark state which initializes 1_000 initial keys in the map,
+     * which map to 1_000 random other keys in the map.
+     */
+    public static class SMRInitBenchmarkState extends SMRBenchmarkState {
+
+        /**
+         * The number of keys in the map.
+         */
+        final int num_keys = 1_000;
+
+        /**
+         * A random for generating random numbers.
+         */
+        Random r = new Random();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4651477)